src/actions/deviceActions.js

```diff
@@ -271,7 +271,12 @@ export function open(deviceInfo) {
                 nbSamples += 1;
                 nbSamplesTotal += 1;
                 const f = Math.min(nbSamplesTotal, samplesPerAverage);
-                if (prevValue !== undefined && value !== undefined) {
+                if (
+                    prevValue !== undefined &&
+                    value !== undefined &&
+                    !Number.isNaN(prevValue) &&
+                    !Number.isNaN(value)
+                ) {
```

Comment:
How about if we use [`Number.isFinite()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite) here? That is a bit shorter and besides handling `undefined` and `NaN`, it also rejects `null` or `Infinity`. Both also should not happen, but I suspect that, just like `NaN`, both would also ruin all future samples.

```suggestion
                if (Number.isFinite(value) && Number.isFinite(prevValue)) {
```

Reply:
I like it. Will add this before releasing v3.1.3.